norns/clocks.md

```diff
@@ -0,0 +1,211 @@
+---
+layout: default
+parent: scripting
+grand_parent: norns
+title: clocks
+nav_order: 3
+permalink: /norns/clocks/
+---
+
+# clocks
+
+the clock library provides a way to create timed function calls: loops, repetition, and delays. synchronization is possible and time base can come from a variety of sources.
+
+## commands
+
+- `id = clock.run( func )` &mdash; start a new clock with function _func_. returns _id_
+- `clock.cancel( id )` &mdash; cancel clock _id_ 
+- `clock.sleep( time )` &mdash; resume in _time_ seconds
+- `clock.sync( beats )` &mdash; resume in _beats_ according to tempo
+- `beats = clock.get_time_beats()` &mdash; returns current time in _beats_
+- `tempo = clock.get_tempo()` &mdash; returns current _tempo_
+- `beat_sec = clock.get_beat_sec()` &mdash; returns length of beat in seconds
+
+## callbacks
+
+these functions can be script defined and are called by system events:
+
+- `clock.transport.start` &mdash; called by transport start
+- `clock.transport.stop` &mdash; called by transport stop
+
+## use
+
+### simple delay
+
+`clock.run` starts a new clock with the function you provide:
+
+```
+function later()
+  clock.sleep(2)
+  print("now awake")
+end
+
+function init()
+  print("starting now")
+  clock.run(later)
+  print("done with init")
+end
+```
+
+when executed, you will see _starting now_, _done with init_, (two seconds delay and then): _now awake_. `clock.run` starts a coroutine with the function `later`. this functions immediately sleeps for two seconds, which means the `init` function resumes.
+   
+### repetition and sync
+
+you can sleep according to units of tempo (beats) by using `clock.sync` instead of `clock.sleep`.
+
+```
+engine.name = 'PolyPerc'
+
+function strum()
+  for i=1,8 do
+    engine.hz(i*100)
+    clock.sync(1/4)
```

Comment:
```suggestion
    clock.sync(1/4)
    engine.hz(i*100)
```

unless we want to show a situation where the first note is un-quantized, this will make sure all notes are synced to 1/16th tick.